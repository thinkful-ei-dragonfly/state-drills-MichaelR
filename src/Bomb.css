.Bomb {
  margin: 20px;
}