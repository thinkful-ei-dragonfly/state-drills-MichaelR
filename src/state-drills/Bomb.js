import React from 'react';
import './Bomb.css';

class Bomb extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      count: 0,
      display: 'tick'
    }
  }

  componentDidMount() {
    this.interval = setInterval(() => {
      let newCount = this.state.count + 1;
      this.setState({
        count: newCount
      });
      if (this.state.count > 7) {
        this.setState({
          display: 'BOOM!!!!'
        })
        clearInterval(this.interval);
      }else if (this.state.count % 2 === 0) {
        this.setState({
          display: 'tick'
        })
      } else {
        this.setState({
          display: 'tock'
        })
      }
    }, 1000);
  }

  componentWillUnmount() {
    clearInterval(this.interval);
  }

  render() {
    return <div className="Bomb">
      <p>{this.state.display}</p>
    </div>
  }
}

export default Bomb;