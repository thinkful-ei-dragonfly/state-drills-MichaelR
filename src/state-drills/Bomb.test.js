import React from 'react';
import renderer from 'react-test-renderer';

import Bomb from './Bomb';

describe('<Bombs />', () => {
  it('renders without crashing [snapshot test]', () => {
    // Render this component as JSON
    const tree = renderer.create(<Bomb />).toJSON();
    // Checks wether this matches the previous snapshot
    // The previous snapshot will be stored in __snapshots__/Bombs .test.js
    expect(tree).toMatchSnapshot();
  })
})