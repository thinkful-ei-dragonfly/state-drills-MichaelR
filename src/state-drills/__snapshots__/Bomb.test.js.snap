// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Bombs /> renders without crashing [snapshot test] 1`] = `
<div
  className="Bomb"
>
  <p>
    tick
  </p>
</div>
`;
